lib/sidekiq/instrument/mixin.rb

```diff
@@ -13,10 +13,17 @@ def worker_dog_options(worker)
     end
 
     def max_retries(worker)
-      retries = worker.class.get_sidekiq_options['retry'] || Sidekiq[:max_retries]
-      return Sidekiq[:max_retries] if retries.to_s.eql?("true")
-      return 0 if retries.eql?("false")
-      retries
+      retries = fetch_worker_retry(worker)
+      case retries.to_s
+      when "true"
+        Sidekiq[:max_retries]
+      when "false"
+        0
+      when ""
+        Sidekiq[:max_retries]
+      else
+        retries
+      end
```

Comment:
Consider 

```suggestion
      case retries.to_s
      when "true", ""
        Sidekiq[:max_retries]
      when "false"
        0
      else
        retries
      end
```

Reply:
But, honestly, we should not be passing in boolean values at all, Sidekiq is clearly written with numbers in mind (and just doesn't validate that)

Reply:
Oh neat, I didn't know that syntax was allowed in Ruby - good shout thanks.

I agree that it's unfortunate we have to handle so many cases because there doesn't really seem to be a good standardized way we're doing it :(